secretsauce/apps/portal/models.py

```diff
@@ -29,86 +29,84 @@ def __str__(self):
         return self.name
 
 class ModelTag(models.Model):
-    name = models.CharField(max_length=200)
+    name = models.CharField(max_length=200, unique=True, primary_key=True)
     prediction_model = models.ManyToManyField(
         PredictionModel,
-        related_name="tags"
+        related_name='model_tags',
     )
-    # TODO: No repeated tags
 
     def __str__(self):
         return self.name
 
-class RequiredHyperparameter(models.Model):
-    """
-    Required hyperparameters to train a defined PredictionModel
-    """
-    prediction_model = models.OneToOneField(
-        PredictionModel, 
-        on_delete=models.CASCADE,
-        related_name='required_hyperparameters'
-    )
-    name = models.CharField(max_length=200)
-    HYPERPARAM_TYPES = (
-        ('INT', "Integer"),
-        ('FP', "Floating point"),
-        ('STR', "String"),
-        ('CAT', "Category")
-    )
-    hyperparameter_type = models.CharField(max_length=3, choices=HYPERPARAM_TYPES)
-
-    def __str__(self):
-        return "%s: %s (%s)" %(self.prediction_model, self.name, self.hyperparameter_type)
-
 class DataBlock(models.Model):
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
     project = models.ForeignKey(
         Project, 
         on_delete=models.CASCADE, 
-        related_name="projects"
+        related_name='data_blocks'
     )
     name = models.CharField(max_length=200)
     upload = models.FileField(upload_to='uploads/')
 
     def __str__(self):
         return "DataBlock: %s" % (self.name) 
 
-class Weights(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    prediction_model = models.ForeignKey(
-        PredictionModel, 
-        on_delete=models.CASCADE, 
-        related_name="weights"
-    )
-    data_block = models.ForeignKey(
-        DataBlock, 
-        on_delete=models.CASCADE,
-        related_name="weights"
-    )
-    upload = models.URLField(max_length = 200) 
-
-    def __str__(self):
-        return "Weight for %s (%s)" % (self.prediction_model, self.data_block)
-
 class ConstraintBlock(models.Model):
+    """A set of constraints"""
+
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    data_block = models.ForeignKey(
-        DataBlock, 
-        on_delete=models.CASCADE, 
-        related_name="datablocks"
+    project = models.ForeignKey(
+        Project,
+        on_delete=models.CASCADE,
+        related_name='constraint_blocks',
     )
```

Comment:
Shouldn't constraint block be linked to datablock? Like different csv files will probably have different items so they should have different constraints too despite being in the same project.

Reply:
I was thinking about the case where, what if two DataBlocks share the same schema, and they actually want to reuse the constraints? Will it be better if we link ConstraintBlocks to Projects, then return an error if they choose wrongly?

Reply:
Okay let's stick to projects then